lib/compiler/src/compiler.rs

```diff
@@ -99,12 +99,14 @@ pub trait Compiler {
         _module: &'module mut CompileModuleInfo,
         _module_translation: &ModuleTranslationState,
         // The list of function bodies
-        _function_body_inputs: &PrimaryMap<LocalFunctionIndex, FunctionBodyData<'data>>,
+        _function_body_inputs: PrimaryMap<LocalFunctionIndex, MiddlewareBinaryReader>,
         _symbol_registry: &dyn SymbolRegistry,
         // The metadata to inject into the wasmer_metadata section of the object file.
         _wasmer_metadata: &[u8],
-    ) -> Option<Result<Vec<u8>, CompileError>> {
-        None
+    ) -> Result<Vec<u8>, CompileError> {
+        Err(CompileError::Codegen(
+            "This backend does not support `experimental_native_compile_module`.".into(),
+        ))
     }
```

Comment:
We return None in propose, so it can fallback to the non-experimental one

Reply:
The API changed to take the function body data by move instead of by reference, so once moved you no longer hold the data to make the fallback call. In its place, a "supports_experimental_native_compile_module" method is provided so that the caller can decide which to use and then move the data once.

Why does it now take by move instead of by reference? Previously the PrimaryMap was from local function index to raw wasm data that was then interpreted, but now it's to a streaming interface. To use a streaming interface you call something like a `get_next()` method which returns different data each time. To do that, the streaming object must be mutating under the hood, so a non-mutating reference is no longer a choice.

It might be possible to make this work with a mutable reference, I didn't try that.